src/MultipartStreamBuilder.php

```diff
@@ -72,6 +72,23 @@ public function __construct($streamFactory = null)
         }
     }
 
+    /**
+     * Add a resource to the Multipart Stream
+     *
+     * @param string|resource|\Psr\Http\Message\StreamInterface $resource
+     *     The filepath, resource or StreamInterface of the data.
+     * @param array $headers
+     *     Additional headers array: ['header-name' => 'header-value'].
+     *
+     * @return MultipartStreamBuilder
+     */
+    public function addData($resource, array $headers = [])
```

Comment:
we discussed making this protected, not public. but i guess public is actually fine.